src/YUILoader.cc

```diff
@@ -175,38 +180,44 @@ void YUILoader::loadRestAPIPlugin( const std::string & wantedGUI, bool withThrea
     yuiMilestone () << "Requested to start http server to control UI." << std::endl;
     if( pluginExists( YUIPlugin_RestAPI ) )
     {
-        // TODO: Do not load unused libraries
-        // Load underlying UI plugin, as test method inherits from it
-        // YUIPlugin_Test uses both libraries to be single point of entry
-        YUIPlugin uiPluginNC( YUIPlugin_NCurses );
-        YUIPlugin uiPluginQT( YUIPlugin_Qt );
-        YUIPlugin uiTestPlugin( YUIPlugin_RestAPI );
-
+        YUIPlugin uiRestPlugin( YUIPlugin_RestAPI );
+        createUIFunction_t createUI = nullptr;
         yuiMilestone () << "User-selected underlying UI-plugin: \"" << wantedGUI << "\"" << std::endl;
-        if ( uiPluginNC.success() && uiPluginQT.success() && uiTestPlugin.success() )
+
+        if (wantedGUI == YUIPlugin_Qt)
         {
-            yuiMilestone () << "Loading http server to control UI." << std::endl;
+            YUIPlugin uiPluginQT( YUIPlugin_Qt );
```

Comment:
What happens when those variables go out of scope? Are the plugins then unloaded in the destructors? Are they not needed anymore? Or should those be member variables to make sure they remain alive until the UI shuts down?

Reply:
To be honest I do not know what exactly happens, but that was already used in the old code (like [here](https://github.com/libyui/libyui/blob/SLE-15-SP1/src/YUILoader.cc#L200)), which is hopefully correct. We just used the same approach. :wink: 

I have never seen any crash or problem regarding that, so I hope it's OK.

Reply:
https://github.com/libyui/libyui/blob/master/src/YUIPlugin.cc#L59

> This intentionally does NOT call unload(): This would be
> counterproductive for almost all use cases of this class.

So the plugin remains loaded; we'd explicitly have to call `unload()` to unload it.
OK then.